### PR TITLE
[161] Location data recommendations and validation. Saving latitude a…

### PR DESCRIPTION
**Pull request by SaidganiUsm**
### Where

[NL-161 Update location inputs to process only correct cities/countries etc](https://jjbcapital.atlassian.net/browse/NL-161)

### What

Autofilling address line in the lot creation stage, fetching latitude and longitude of a provided address, saving this geolocation, and using it in lot profile for map.

### How

I used Goeocode API from Google MapsApi and used a leaflet package for showing maps based on latitude and longitude.

### Affected Area

Auctionify.API
Auctionify.Application
Auctionify.Entitie
Auctionify.Infrastructure

### PR Checklist

_Build_

- [x] Build Succeeded

_Unit Tests_

- [x] X Tests Passed, Y Ignored, Z Failed

### CODE REVIEW

**Review by khasanrashidov — DISMISSED** [review]

File: src/Auctionify.Application/ConfigureServices.cs
Comment (TsvirkoNikita) [inline src/Auctionify.Application/ConfigureServices.cs:1]: Remove unused usings

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]
- When editing a lot, lot update request fails without any error message for user. The reason for the error is missing Latitude and Longitude fields in the request:
image
Fix this bug and fix popup with any errors showing.
- Another issue related to Country and City fields. You still can put anything to it after you enter Address field:
image
I suggest making these fields disabled as they are filled in automatically after entering the address anyway

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]
- If scroll page down before open location popup, Google suggestions displaying too up
- Country and City inputs disabled for input, but still looks like active. It should be displayed as disabled with gray background
image

**Comment by SaidganiUsm** [comment]
Unfortunately, it was not possible to correct the location of the location recommendations element

**Review by TsvirkoNikita — APPROVED** [review]

**Review by khasanrashidov — APPROVED** [review]

**Pull request merged**